src/sourcemaps/utils/sdk-version.ts

```diff
@@ -0,0 +1,213 @@
+// @ts-ignore - clack is ESM and TS complains about that. It works though
+import clack from '@clack/prompts';
+import chalk from 'chalk';
+import { minVersion, satisfies } from 'semver';
+import {
+  abortIfCancelled,
+  getPackageDotJson,
+  getPackageVersion,
+  hasPackageInstalled,
+  installPackage,
+} from '../../utils/clack-utils';
+
+import * as Sentry from '@sentry/node';
+
+const MINIMUM_DEBUG_ID_SDK_VERSION = '7.47.0';
+
+const SENTRY_SDK_PACKAGES = [
+  // SDKs using other framework SDKs need to be checked first
+  '@sentry/gatsby',
+  '@sentry/nextjs',
+  '@sentry/remix',
+  '@sentry/sveltekit',
+
+  // Framework SDKs
+  '@sentry/angular',
+  '@sentry/angular-ivy',
+  '@sentry/ember',
+  '@sentry/react',
+  '@sentry/svelte',
+  '@sentry/vue',
+  '@sentry/serverless',
+
+  // Base SDKs
+  '@sentry/browser',
+  '@sentry/node',
+];
+
+/**
+ * Check for a minimum SDK version and prompt the user to upgrade if necessary.
+ * We distinguish between 4 cases here:
+ *
+ * 1. Users didn't install any SDK yet
+ *    -> We tell them to install an SDK and then continue with the wizard
+ * 2. Users installed an SDK in the range >=7.47.0
+ *    -> All good, no need to do anything!
+ * 3. Users installed an SDK in the range >=7.0.0 <= 7.46.0
+ *    -> We ask if they want to auto-update to the latest version
+ * 4. Users installed an SDK in the range <7.x
+ *    -> We tell users to manually upgrade (migrate between majors)
+ */
+export async function ensureMinimumSdkVersionIsInstalled(): Promise<void> {
+  const packageJson = await getPackageDotJson();
+
+  const installedSdkPackage = SENTRY_SDK_PACKAGES.find((sdkPackage) =>
+    hasPackageInstalled(sdkPackage, packageJson),
+  );
+
+  // Case 1:
+  if (!installedSdkPackage) {
+    return await handleNoSdkInstalled();
+  }
+
+  const installedSdkVersionOrRange =
+    getPackageVersion(installedSdkPackage, packageJson) || '';
```

Comment:
Falling back to empty string is pretty hard to reason about when looking at this code. What is conna happen inside `satisfies` below?

Can we somehow either unify the `hasPackageInstalled` and `getPackageVersion` call so we don't have to do the fallback logic or explicitly bail when the version is undefined?

Reply:
I refactored this a little because I'll need to extract it anyway for the SvelteKit/NextJS check later. Typing still isn't perfect but right now it's at least reduced to the typing problem coming from `semver` rather than ours 😅